cognite_toolkit/_cdf_tk/utils/fileio/_writers.py

```diff
@@ -44,18 +45,22 @@ def __init__(
         self.max_file_size_bytes = max_file_size_bytes
         self._file_count_by_filename: dict[str, int] = Counter()
         self._writer_by_filepath: dict[Path, T_IO] = {}
+        self._lock = threading.RLock()  # Use RLock to allow recursive locking
```

Comment:
I don't understand why you need a reentrant lock here, could you elaborate?
```suggestion
        self._lock = threading.Lock()
```

Reply:
I guess it is overkill as the lock is only used internally the FileWriter.